ncm-grub/src/main/perl/grub.pm

```diff
@@ -74,6 +76,70 @@ sub grubbyArgsOptions {
 
 }
 
+sub Configure_password {
+    my ($self, $config, $editor) = @_;
+
+    my $password = undef;
+    my $passwordpath = $self->prefix . "/password";
```

Comment:
it is much better to use `$config->getTree($self->prefix);` (or `$config->getElement($self->prefix())->getTree()` for old CCM versions, and you can use the element and data as a hashref (so in Configure you could do
```perl
my $tree = $config->getTree($self->prefix);
...
$self->Configure_password($tree->{password}, $editor) if $tree->{password};
```
and then you can use the passed password tree as regular hashref
in particular, `$config->getElement("$passwordpath/enabled")->getValue()` below returns the stirng 'false' for the boolean `false`, so it is always true. with getTree, booleans are converted in 0/1 so they can be used as booleans.